node/chainSimulator/chainSimulator_test.go

```diff
@@ -380,3 +387,76 @@ func TestSimulator_SendTransactions(t *testing.T) {
 
 	chainSimulatorCommon.CheckGenerateTransactions(t, chainSimulator)
 }
+
+func TestSimulator_SentMoveBalanceNoGasForFee(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+	chainSimulator, err := NewChainSimulator(ArgsChainSimulator{
+		BypassTxSignatureCheck: true,
+		TempDir:                t.TempDir(),
+		PathToInitialConfig:    defaultPathToInitialConfig,
+		NumOfShards:            3,
+		GenesisTimestamp:       startTime,
+		RoundDurationInMillis:  roundDurationInMillis,
+		RoundsPerEpoch:         roundsPerEpoch,
+		ApiInterface:           api.NewNoApiInterface(),
+		MinNodesPerShard:       1,
+		MetaChainMinNodes:      1,
+	})
+	require.Nil(t, err)
+	require.NotNil(t, chainSimulator)
+
+	defer chainSimulator.Close()
+
+	wallet0, err := chainSimulator.GenerateAndMintWalletAddress(0, big.NewInt(0))
+	require.Nil(t, err)
+
+	ftx := transaction.FrontendTransaction{
+		Nonce:     0,
+		Value:     "0",
+		Sender:    wallet0.Bech32,
+		Receiver:  wallet0.Bech32,
+		Data:      []byte(""),
+		GasLimit:  50_000,
+		GasPrice:  1_000_000_000,
+		ChainID:   configs.ChainID,
+		Version:   1,
+		Signature: "010101",
+	}
+
+	txArgs := &external.ArgsCreateTransaction{
+		Nonce:            ftx.Nonce,
+		Value:            ftx.Value,
+		Receiver:         ftx.Receiver,
+		ReceiverUsername: ftx.ReceiverUsername,
+		Sender:           ftx.Sender,
+		SenderUsername:   ftx.SenderUsername,
+		GasPrice:         ftx.GasPrice,
+		GasLimit:         ftx.GasLimit,
+		DataField:        ftx.Data,
+		SignatureHex:     ftx.Signature,
+		ChainID:          ftx.ChainID,
+		Version:          ftx.Version,
+		Options:          ftx.Options,
+		Guardian:         ftx.GuardianAddr,
+		GuardianSigHex:   ftx.GuardianSignature,
+	}
+
+	shardFacadeHandle := chainSimulator.nodes[0].GetFacadeHandler()
```

Comment:
why not test that the direct send transaction returns an error?

Reply:
done, changed